torch/fft/__init__.py

```diff
@@ -9,18 +9,233 @@
+    n (int, optional): FFT length. If given, the input will either be zero-padded
+        or trimmed to this length before taking the FFT.
+    dim (int, optional): The dimension along which to take the one dimensional IFFT.
+    norm (str, optional): Normalization mode:
+        ``"forward"`` - the forward transform (:func:`~torch.fft.rfft`) is normalized by ``1/n``,
+        ``"backward"`` - the backward transform (:func:`~torch.fft.irfft`) is normalized by ``1/n``,
+        ``"ortho"`` - both transform directions are normalized by ``1/sqrt(n)`` (making it orthonormal).
+        Default is ``"backward"``.
 
 Example::
 
-    >>> t = torch.randn(4, dtype=torch.complex128)
+    >>> import torch.fft
+    >>> t = torch.arange(4)
     >>> t
-    tensor([-1.1364-0.5694j, -0.6637+0.9987j, -1.0102-0.4383j,  0.3017+0.9371j], dtype=torch.complex128)
+    tensor([0, 1, 2, 3])
+    >>> torch.fft.rfft(t)
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j])
+
+    Compare against the full output from :func:`torch.fft.fft`:
+
     >>> torch.fft.fft(t)
-    tensor([-2.5086+0.9281j, -0.0646+0.8343j, -1.7846-2.9435j, -0.1878-1.0965j], dtype=torch.complex128)
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j, -2.-2.j])
+
+    Notice that the symmetric element ``T[-1] == T[1].conj()`` is ommitted.
+    At the Nyquist freqency ``T[-2] == T[2].conj() == T[2]`` is given but must
+    always be purely real.
+""")
+
+irfft = _add_docstr(_fft.fft_irfft, r"""
+irfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the inverse of :func:`torch.fft.rfft`.
+
+:attr:`input` is a one-sided hermitian signal in the fourrier domain, as
+produced by :func:`torch.fft.rfft`. By the hermitian property, the time-domain
+signal will be purely real.
+
+Note that some input frequencies must be purely real to satisfy the hermitian
+property. In these cases the imaginary component will be ignored.
+
+Args:
+    input (Tensor): the input tensor representing a half-hermitian signal
+    n (int, optional): Output signal length. This determines the length of the
+        output signal. If given, the input will either be zero-padded or trimmed to this
+        length before taking the IFFT. Defaults to odd output: ``n=input.size(dim) * 2 - 1``.
+    dim (int, optional): The dimension along which to take the one dimensional IFFT.
+    norm (str, optional): Normalization mode:
+        ``"forward"`` - the forward transform (:func:`~torch.fft.rfft`) is normalized by ``1/n``,
+        ``"backward"`` - the backward transform (:func:`~torch.fft.irfft`) is normalized by ``1/n``,
+        ``"ortho"`` - both transform directions are normalized by ``1/sqrt(n)`` (making it orthonormal).
+        Default is ``"backward"``.
+
+Example::
+
+    >>> import torch.fft
+    >>> t = torch.arange(4)
+    >>> t
+    tensor([0, 1, 2, 3])
+    >>> T = torch.fft.rfft(t)
+    >>> T
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j])
+
+    Without specifying the output length to :func:`torch.fft.irfft`, the output
```

Comment:
Should this be a note and then the example can have a comment like "passing signal length to support round-trips"?